test/integration/conftest.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+from __future__ import annotations
+
+import functools
+import logging
+import subprocess
+from typing import Callable, Generator
+
+import jubilant
+import minio
+import pytest
+
+
+@pytest.fixture
+def juju() -> Generator[jubilant.Juju, None, None]:
+    """Make a Juju model with the tracing part of COS ready."""
+    with jubilant.temp_model() as j:
+        deploy_tempo(j)
+        deploy_tempo_worker(j)
+        j.deploy('minio', config={'access-key': 'accesskey', 'secret-key': 'mysoverysecretkey'})
```

Comment:
I'd prefer to see `juju.deploy()` instead of `j.deploy()` -- it reads more clearly. I know some tools complain about the shadowed name, but that's kind of an issue with how Pyright's fixture naming works. So I'd just `noqa` or turn off that warning and rename this to `juju`.